indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -306,9 +398,16 @@ object Asn1EncodableSerializer : KSerializer<Asn1Element> {
 /**
  * ASN.1 structure. Contains no data itself, but holds zero or more [children]
  */
-sealed class Asn1Structure(tag: Tag, children: List<Asn1Element>?) :
-    Asn1Element(TLV(tag, byteArrayOf()), children) {
+sealed class Asn1Structure(tag: Tag, children: List<Asn1Element>?, sort: Boolean = false) :
+    Asn1Element(TLV(tag, byteArrayOf()), if (!sort) children else children?.sortedBy { it.tag }) {
 
+    /**
+     * Whether this structure sorts child nodes or keeps them as-is.
+     * This **should** be true for SET and SET OF, but is set to false for SET and SET OF elements parsed
+     * from DER-encoded structures, because this has a chance of altering the structure for non-conforming DER-encoded
```

Comment:
name & shame some examples 😈

Reply:
one comes to mind, but i am confident neither of us can even fathom how much garbage is out there. 

For the record: Apple is one offender that comes to mind

Reply:
i meant give it as an example in the docs for why this is needed in practice, otherwise it's unintuitive why this would account for non-compliant code

Reply:
the comment already explains why this counter-intuitive flag is required. i am against pointing fingers in the docs here, because our implementation is also not feature-complete and allows for generating non-compliant structures, if we apply the SET tag to custom structures